ENCD-6262-elements-cloning-and-elements-mappings-display

forresttanaka opened the pull request:
* I removed the `analysisObjects` collapsed-state initialization as it never seemed to come up with results, and the rules for the default collapse/expanded states of the various file tables has changed in this ticket anyway.
* Related, I removed the collapsed states from the raw data table and the raw sequencing table so that FileTable can coordinate all the tables for the new collapse rules.

forresttanaka commented:
~https://encd-6262-fcs-files-pv2-fytanaka.demo.encodedcc.org/search/?type=FunctionalCharacterizationSeries&related_datasets.elements_cloning=*&limit=all~

No `elements_mappings` exist in production data. I have a couple in local test data.

emro commented on src/encoded/static/components/dataset.js:
```diff
@@ -2868,12 +2870,15 @@ export const SeriesComponent = ({
                     supplementalShortLabels={supplementalShortLabels}
                     showReplicateNumber={false}
                     hideControls={!METADATA_SERIES_TYPES.includes(context['@type'][0])}
-                    collapseNone
```
This is funny, I guess this isn't used anywhere. Maybe it's left from some previous change that did need it.

forresttanaka replied:
The rules about collapsing have changed in this ticket, so that’s why this went away.

emro commented on src/encoded/static/components/dataset.js:
```diff
@@ -3164,13 +3173,119 @@ DiseaseSeries.contextTypes = {
+            const cloningAndAnalysisFilePaths = _.intersection(dataset.files, analysisFilesPaths);
+            files = cloningAndAnalysisFilePaths
+                .map((filePath) => allFiles.find((file) => file['@id'] === filePath))
+                .filter((file) => file);
```
What does this do? (`.filter((file) => file)`) I don't know if I've seen this before.

forresttanaka replied:
It’s a local version of _.compact(). It just removes any falsy entries in the array, and easier than wrapping the whole RHS in _.compact().

emro reviewed (changes requested):
The code all looks good but the conflicting file needs to be resolved.

forresttanaka commented:
I updated https://encd-6262-fcs-files-qa1-fytanaka.demo.encodedcc.org/ after rebasing.

emro reviewed (dismissed):
Looks good!

forresttanaka committed:
ENCD-6262-fcs-files

Show non-default files in related datasets of functional characterization series.

forresttanaka committed:
ENCD-6262-fcs-files

Add a note to the series batch download modal about downloading cloning and mappings files from their sections, not from the main download.

forresttanaka committed:
ENCD-6262-fcs-files

forresttanaka committed:
ENCD-6262-fcs-files

Grab Idan’s changes to allow elements_cloning and elements_mappings files in the new series_files property.

forresttanaka committed:
ENCD-6262-fcs-files

Correct typo.

forresttanaka committed:
ENCD-6262-fcs-files

Revert note about not being able to download elements_cloning and elements_mappings in the file gallery.

emro commented on src/encoded/static/components/batch_download/cloning_mappings.js:
```diff
@@ -0,0 +1,72 @@
+            {
+                id: 'all-files',
+                label: 'Download files',
+                title: `All experiment${this._hasAnalyses ? `${uc.rsquo}s analyses` : ''} files`,
```
Should this be "all experiments' analysis files"?
![Screen Shot 2022-02-03 at 10 39 18 AM](https://user-images.githubusercontent.com/2887343/152407648-b66642a3-01e2-4569-a19d-6296cbb79acf.png)

emro reviewed (dismissed):
Maybe one re-wording thing, but code all looks great.

forresttanaka committed:
ENCD-6262-fcs-files

Account for analyses property to FileGallery to always be an array instead of possibly being null.

forresttanaka committed:
ENCD-6262-fcs-files

A high-risk fix for series analyses not getting displayed (second issue Ingrid found in the ticket). Just have to give it a try.

forresttanaka committed:
ENCD-6262-fcs-files

One more field needed.

forresttanaka committed:
ENCD-6262-fcs-files

Use @ids of analyses now that we get the entire objects.

forresttanaka committed:
ENCD-6262-fcs-files

We could get analyses @ids or whole objects now. This is getting extremely messy and brittle.

forresttanaka committed:
ENCD-6262-fcs-files

Now mutating files to normalize them to have @ids in their files.series.analyses.

forresttanaka committed:
ENCD-6262-fcs-files

Delete unneeded line.

forresttanaka committed:
ENCD-6262-fcs-files

forresttanaka committed:
ENCD-6262-fcs-files

Weird merge conflict. Trying to restore file inserts that I had.

forresttanaka committed:
ENCD-6262-fcs-files

Correct download text.

forresttanaka committed:
ENCD-6262-fcs-files

Restore file.json back to what’s on dev. I’ll try to patch this branch’s original changes in the next commit.

forresttanaka committed:
ENCD-6262-fcs-files

Got things back to the way they’re supposed to be.

emro reviewed (approved):
Code changes look good. This is a very complicated ticket.
